simba/costs.py

```diff
@@ -101,20 +101,42 @@ def calculate_costs(c_params, scenario, schedule, args):
 
     # CHARGING INFRASTRUCTURE
     cs = schedule.scenario["components"]["charging_stations"]
-    # depot charging stations - each charging bus generates one CS
-    for csID in cs.values():
-        if csID["type"] == "deps":
-            costs["c_cs"] += c_params["cs"]["capex_deps_per_kW"] * csID["max_power"]
-    # opportunity charging stations - nr of CS depend on max nr of simultaneously occupied CS
     for gcID in gcs:
-        if schedule.stations[gcID]["type"] == "opps":
+        # depot charging stations - each charging bus generates one CS
+        if schedule.stations[gcID]["type"] == "deps":
+            if schedule.stations[gcID]["n_charging_stations"] is not None:
+                # get nr of CS from electrified stations data
+                n_cs = min(sum(gcID in i for i in cs.keys()),
+                           schedule.stations[gcID]["n_charging_stations"])
+                costs["c_cs"] += (
+                        c_params["cs"]["capex_deps_per_kW"] * n_cs *
+                        # get max. defined power for this deps or else max. power at deps in general
+                        # ToDo: Calculation does not work!
+                        #  max("cs_power_deps_depb", "cs_power_deps_oppb") always takes value of
+                        #  cs_power_deps_oppb, since only the strings are compared.
+                        #  Instead it should compare both entries and return max value
+                        #  if one or both exist.
+                        schedule.stations[gcID].get(max("cs_power_deps_depb", "cs_power_deps_oppb"),
+                                                    max(vars(args)["cs_power_deps_depb"],
+                                                        vars(args)["cs_power_deps_oppb"])))
+            else:
+                # check for electrified stations without a predefined number of CS in list of all CS
+                for csID in cs.values():
+                    if gcID == csID["parent"]:
```

Comment:
you are not iterating over keys, why is this called csID? In general, I would suggest to create a sublist of CS at this location: `cs_at_station = [c for c in cs.keys if cs.parent == gcID]`. This also simplifies line 109.